### PR TITLE
Add configure option --disable-libdl

### LINKED ISSUE #907: 5.0.0: Error in dlsym(RTLD_NEXT, "pthread_create")
jemalloc 4.5.0 worked, but not 5.0.0.

```
$ LD_DEBUG=files ldd mpv 2>&1|grep libjemalloc
      3778:	file=libjemalloc.so.2 [0];  needed by /lib64/libavcodec.so.57 [0]
      3778:	file=libjemalloc.so.2 [0];  generating link map
	libjemalloc.so.2 => /lib64/libjemalloc.so.2 (0x00007f5664908000)
```
```
$ ./mpv
<jemalloc>: Error in dlsym(RTLD_NEXT, "pthread_create")
Aborted
```

Fedora 25 x86_64, glibc-2.24-6.fc25.

### PR DESCRIPTION
**Pull request by interwq**
This makes it possible to build full static binary.

This resolves #907

### CODE REVIEW

**Comment by interwq** [comment]
Will rebase past #1242

File: configure.ac [hunk 1]
```diff
@@ -1495,6 +1495,21 @@ AC_SUBST([jemalloc_version_bugfix])
...
+AC_ARG_ENABLE([libdl],
+  [AS_HELP_STRING([--disable-libdl],
+  [Do not use libdl])],
+[if test "x$enable_libdl" = "xno" ; then
+  enable_libdl="0"
+else
+  enable_libdl="1"
+fi
+],
+[enable_libdl="0"]
```
Comment (interwq) [inline configure.ac]: CI only. Will change back.

**Comment by davidtgoldblatt** [comment]
Would it be possible to detect libdl at runtime (say, by declaring as weak symbols the things we want)? Every new configure flag is a personal affront to my dignity.

**Comment by interwq** [comment]
Chatted with @davidtgoldblatt offline. This is to provide a way to not link against -ldl. Ready for review now.

**Comment by interwq** [comment]
All tests passed. Will change the default setting and rerun again to make sure.

**Comment by davidtgoldblatt** [comment]
Let's add a --disable-libdl option to CI, too?

**Comment by interwq** [comment]
@davidtgoldblatt: We explicitly force lazy lock on FreeBSD, which can't work w/o libdl. I'm actually not too worried about testing the option w/ CI, the scope it affects is very limited (a single `#ifdef`pretty much, which likely won't be touched anytime soon). Does that sound reasonable?

**Comment by davidtgoldblatt** [comment]
We don't have to test it on freebsd; but we should check to make sure that we don't see something unexpected start happening with our interception otherwise.

I'm just suggesting adding `--disable-libdl` to  `https://github.com/jemalloc/jemalloc/blob/dev/scripts/gen_travis.py#L47`

**Comment by interwq** [comment]
Sounds good. Just added to Travis.

**Pull request merged**